cloudinit/util.py

```diff
@@ -175,8 +175,16 @@ def fully_decoded_payload(part):
         cte_payload, bytes
     ):
         charset = part.get_charset()
-        if charset and charset.input_codec:
-            encoding = charset.input_codec
+
+        # TODO: Mypy doesn't like the following code because `input_codec`
```

Comment:
+1, it also doesn't appear to be [slated for removal in the near future](https://docs.python.org/3.14/whatsnew/3.14.html), so this should be fine. It also doesn't look like a trivial fix so lets keep an eye out in case upstream Python decides that this has to go.